admin/spam/views.py

```diff
@@ -1,47 +1,106 @@
 from django.shortcuts import render
-from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
 from django.contrib.auth.decorators import login_required
+from django.views.generic import FormView, ListView
+from django.utils.decorators import method_decorator
+from django.core.urlresolvers import reverse
+from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 
 from .serializers import serialize_comment
+from .forms import ConfirmForm
 
 
-def get_spam_list():
-    query = (
-        Q('reports', 'ne', {}) &
-        Q('reports', 'ne', None)
-    )
-    return Comment.find(query).sort('date_created')
+class SpamList(ListView):
+    template_name = 'spam/spam.html'
+    paginate_by = 10
+    paginate_orphans = 1
+    ordering = 'date_created'
+    context_object_name = 'Spam'
 
+    def __init__(self):
+        self.status = str(Comment.FLAGGED)
+        super(SpamList, self).__init__()
 
-@login_required
-def spam_list(request):
-    paginator = Paginator(get_spam_list(), 10)
-
-    page_number = request.GET.get('page', 1)
-    try:
-        page = paginator.page(page_number)
-    except PageNotAnInteger:
-        page = paginator.page(1)
-    except EmptyPage:
-        page = paginator.page(paginator.num_pages)
-    context = {
-        'spam': map(serialize_comment, page),
-        'page': page,
-        'page_number': page_number,
-    }
-    return render(request, 'spam/spam.html', context)
+    def get_queryset(self):
+        self.status = self.request.GET.get('status', u'1')
+        query = (
+            Q('reports', 'ne', {}) &
+            Q('reports', 'ne', None) &
+            Q('spam_status', 'eq', int(self.status))
+        )
+        return Comment.find(query).sort(self.ordering)
 
+    def get_context_data(self, **kwargs):
```

Comment:
You generally want to call super in any CBV defined method unless you have a specific reason not to. There's so much multiple inheritance that you could lose functionality that you never even knew was included.

In this situation you probably want to do whatever you're doing that's different than the base method in `MultipleObjectMixin` and `ContextMixin` and then splat your context to the super call like they do at the end of `MultipleObjectMixin.get_context_data()` https://ccbv.co.uk/projects/Django/1.9/django.views.generic.list/ListView/#get_context_data-MultipleObjectMixin